client/packages.go

```diff
@@ -176,8 +183,9 @@ func (client *Client) Find(opts *FindOptions) ([]*Snap, *ResultInfo, error) {
 	case opts.Private:
 		q.Set("select", "private")
 	}
-	if opts.Section != "" {
-		q.Set("section", opts.Section)
+
+	if section != "" || (section == "" && query == "") {
```

Comment:
This I don't like. The store has this wacky behaviour to support things in a backwards-compatible way, but we have no such problem. So why do this?